bim2sim/elements/bps_elements.py

```diff
@@ -1751,6 +1751,23 @@ def _get_building_name(self, name):
             bldg_name = 'Building'
         return bldg_name
 
+    def _get_number_of_storeys(self, name):
+        return len(self.storeys)
+
+    def _get_avg_storey_height(self, name):
+        """Calculates the average height of all storeys."""
+        storey_height_sum = 0
+        for storey in self.storeys:
+            if storey.height:
+                height = storey.height
+            elif storey.gross_height:
+                height = storey.gross_height
+            elif storey.net_height:
+                height = storey.net_height
+            storey_height_sum += height
```

Comment:
please add an option for cases, where none of these conditions are met (add an else statement).

Reply:
improved this with last commit